### ISSUE TITLE
Determine the file type based on the HTTP Content-Type header

### ISSUE THREAD

**Issue by kdekooter** [danfickle/openhtmltopdf#667]
I am trying to load this (svg) file: https://app.metronome.nl/api/document-view/e1bcaee7-8616-499e-9c2e-f6f5161f6436. In a browser it displays fine. However openpdftohtml emits the following warnings:

```
com.openhtmltopdf.load WARNING:: Unrecognized image format for: https://app.metronome.nl/api/document-view/e1bcaee7-8616-499e-9c2e-f6f5161f6436
com.openhtmltopdf.exception WARNING:: Can't read image file; unexpected problem for URI 'https://app.metronome.nl/api/document-view/e1bcaee7-8616-499e-9c2e-f6f5161f6436'
```
After some experimentation I discovered that appending an `.svg` extension fixed things. Looks like the URL needs to end with `.svg` to be recognized properly...

Suggestion: determine the file type based on the HTTP Content-Type header.

**Comment by kdekooter** [comment]
Looks like the URL needs to end with `.svg` to be recognized properly...

**Comment by syjer** [comment]
hi @kdekooter  you are right, currently for svg and pdf images, it use the file extension:

https://github.com/danfickle/openhtmltopdf/blob/d20825b6984d41704264f265efb05f445e559295/openhtmltopdf-pdfbox/src/main/java/com/openhtmltopdf/pdfboxout/PdfBoxReplacedElementFactory.java#L65


I'm wondering how we could add the support for reading http headers or doing content type check (if loaded from the filesystem or others protocols). Maybe the simplest way would be to simply load the resource as a binary data and do some file header check...

**Comment by kdekooter** [comment]
One way to obtain the http headers - without loading the actual file - would be to do a HEAD request. My sample URL would return the following header: `content-type  image/svg+xml`.

Another way would be to make the `NaiveUserAgent` a little less naive: instead of loading a remote resource with `URL.openStream` it could use an HttpClient for loading. That way you can read the http headers and the (binary) payload in one go.

**Comment by syjer** [comment]
Yes, the main issue is how to fit this new mechanism while keeping the retrocompatibility. Additionally,  http is not the only supported protocol, so we will need some kind of abstraction :).

**Comment by danfickle** [comment]
There is [URLConnection::getContentType](https://docs.oracle.com/javase/8/docs/api/java/net/URLConnection.html#getContentType--)

However, SVGs can be insecure. Therefore, we need to be very careful with adding a change that would allow any image url to pull in insecure content (only determined at runtime based on the returned content-type).

And what if we are dealing with a non-http url?

> Yes, the main issue is how to fit this new mechanism while keeping the retrocompatibility.

I agree. This will be the major issue.

**Comment by kdekooter** [comment]
> However, SVGs can be insecure

SVGs loaded from the file system can also pull in insecure content. Isn't that the concern of the developer using this library and not the library's concern?
